test/sys/test_wait.rs

```diff
@@ -37,6 +37,19 @@ fn test_wait_exit() {
     }
 }
 
+#[test]
+fn test_waitstatus_pid() {
+    let _m = ::FORK_MTX.lock().expect("Mutex got poisoned by another test");
+
+    match fork().unwrap() {
+        Child => {},
```

Comment:
Like that, but worse because the parent is a multi-threaded process.  In a multi-threaded process, even memory allocation could theoretically deadlock, so until the child execs, it is limited to only calling async-signal-safe functions, which are a pretty restrictive set.